agent/src/softwarecontaineragent.h

```diff
@@ -226,7 +228,7 @@ class SoftwareContainerAgent : protected softwarecontainer::JSONParser
     // List of containers in use
     std::map<ContainerID, SoftwareContainerPtr> m_containers;
     // List of pre-loaded containers
-    std::vector<SoftwareContainerPtr> m_preloadedContainers;
+    std::vector<std::pair<ContainerID, SoftwareContainerPtr>> m_preloadedContainers;
```

Comment:
Yes, so the reason is that I need to store the ContainerID but I never want to lookup values. The only functionality that is needed is to pop from the collection. Then maybe a stack would be better.